### PR TITLE
Bdfs

### PR DESCRIPTION
**Pull request by shobs55**

### CODE REVIEW

File: src/bdhost/Util.cpp [hunk 1]
```diff
@@ -77,25 +81,33 @@ namespace bdhost
...
+    json["size"] = Json::Value::UInt(size + reservedSpace);
 
     std::ofstream out(RESERVED_FILE);
-		out << json.toStyledString();
-		out.close();
+    out << json.toStyledString();
+    out.close();
+
+    std::string reserve_id = uuidgen();
+    mkdir(reserve_id.c_str(), 0755);
```
Comment (renqiz) [inline src/bdhost/Util.cpp]: This should go to the storage folder. Current working directory may be /usr/bin.

File: src/bdfs/Options.cpp [hunk 2]
```diff
@@ -95,8 +96,74 @@ namespace dfs
...
+      buffer = buf;
+      offset = size;
+      size += BUFSIZ;
+    }
+
+    fclose(file);
+
+    Json::Reader reader;
+    Json::Value json;
+    if (!reader.parse(buffer, offset + bytes, json, false) ||
```
Comment (renqiz) [inline src/bdfs/Options.cpp]: buffer is never freed

File: src/bdfs/bdfs.conf [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "kademlia" : [ "http://localhost:7800" ],
```
Comment (renqiz) [inline src/bdfs/bdfs.conf]: Talk to victor for public kademlia address in US. We should include that when we create installation package for client. (Better to have a build flag to specify something else for test builds, especially for Shanghai)

File: src/bdfs/Options.cpp [hunk 4]
```diff
@@ -98,6 +98,26 @@ namespace dfs
     exit(format == NULL ? 0 : 1);
   }
 
+  uint64_t Options::ParseSizeStr(const std::string & ssize)
```
Comment (renqiz) [inline src/bdfs/Options.cpp]: We should still allow user to specify the value without the unit

Comment (shobs55) [reply src/bdfs/Options.cpp]: bdfs drive can read command line arg in bytes with parameter -s or can read UINT or String representation from conf json file.

**Pull request merged**